### ISSUE TITLE
False positive for 930120 with json.profile

### ISSUE THREAD

**Issue by S0obi** [coreruleset/coreruleset#4471]
### Description

Rule 930120 will incorrectly block json with profile key.

### How to reproduce the misbehavior (-> curl call)

```bash
curl -sS -H "Content-Type: application/json" -d '{"profile":"test"}' https://example.com
```

Unfortunately CRS sandbox is not supporting newest version and will not show this FP.

### Logs

```
ModSecurity: Warning. Matched "Operator `PmFromFile' with parameter `lfi-os-files.data' against variable `ARGS_NAMES:json.profile' (Value: `json.profile' ) [file "/etc/nginx/modsec/coreruleset-main/rules/REQUEST-930-APPLICATION-ATTACK-LFI.conf"]
[line "99"]
[id "930120"]
[rev ""]
[msg "OS File Access Attempt"]
[data "Matched Data: .profile found within ARGS_NAMES:json.profile: json.profile"]
[severity "2"]
[ver "OWASP_CRS/4.24.0-dev"]
[maturity "0"]
[accuracy "0"]
[tag "application-multi"]
[tag "language-multi"]
[tag "platform-multi"]
[tag "attack-lfi"]
[tag "paranoia-level/1"]
[tag "OWASP_CRS"]
[tag "OWASP_CRS/ATTACK-LFI"]
[tag "capec/1000/255/153/126"]
[hostname "example.com.com"]
[uri "/services/logs"]
[unique_id "177142249330.968256"]
[ref "o4,8v0,12t:utf8toUnicode,t:urlDecodeUni,t:normalizePathWin"]
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:BLOCKING_INBOUND_ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/modsec/coreruleset-main/rules/REQUEST-949-BLOCKING-EVALUATION.conf"]
[line "222"]
[id "949110"]
[rev ""]
[msg "Inbound Anomaly Score Exceeded (Total Score: 5)"]
[data ""]
[severity "0"]
[ver "OWASP_CRS/4.24.0-dev"]
[maturity "0"]
[accuracy "0"]
[tag "anomaly-evaluation"]
[tag "OWASP_CRS"]
[hostname "example.com.com"]
[uri "/services/logs"]
[unique_id "177142249330.968256"]
[ref ""]
```

### Your Environment

- CRS version : v4.24.0-dev
- Paranoia level setting : PL1
- ModSecurity version : latest
- Web Server and version : nginx latest
- Operating System and version: Ubuntu 22.04

### Confirmation

[X] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by airween** [comment]
Hi @S0obi,

thanks for this report.

Yes, unfortunately this is a known issue - when the engine expands the JSON nodes into `ARGS`, it uses a "flat" naming structure. The ROOT key name will be `json`, it uses the `.` (dot) character for the notation, and then add the keys. This is why the `{"profile":"anything"}` will appear as `json.profile`, and CRS catches `.profile`.

The only thing that you can do is to create an exclusion against this unexpected behavior.

**Comment by S0obi** [comment]
Hello @airween, thanks for the explanation, it makes more sense.

I reviewed https://github.com/coreruleset/coreruleset/blob/main/rules/lfi-os-files.data (used by 930120) and it seems that ".profile" may be the most probable "problematic" keyword, as it's a english term (maybe similar situation with .secrets and .tools).

Should we consider to remove .profile from the list ? Not sure if other people encountered this one before me.

**Comment by airween** [comment]
Hi @S0obi,

> Should we consider to remove .profile from the list ?

That would be a solution, or move that keyword/rule to higher PL.

We should discuss that on next monthly chat.

**Comment by EsadCetiner** [comment]
We could handle this edge case by adding `SecRuleUpdateTargetById 930120 !ARGS_NAMES:json.profile` to the recently created `REQUEST-999-COMMON-EXCEPTIONS-AFTER.conf`. I don't think we need to disable this entry or move it to PL-2 if this is the only false positive with `.profile`
